Dp 10

ItsRizee opened the pull request.

ItsRizee committed:
Added: Add product drawer
Added: 404 and 500 page
Added: global user state with zustand
Added: Standart layout for all pages

ItsRizee committed:
Added: Add product drawer
Added: 404 and 500 page
Added: global user state with zustand
Added: Standart layout for all pages

ItsRizee committed:
Fixed: Navbar rendering
Fixed: listeners for user's data change

ItsRizee committed:
Added: Catalog
Added: Displaying craftsman on home page
Added: About us page
Updated: removed static info from Product card and Avatar icon

ItsRizee committed:
Added: Product page
Added: owner, tags and createdDate fields to product class
Fixed: rendering of catalog page
Updated: Product card's details and made it a button

ItsRizee committed:
Merge remote-tracking branch 'origin/main'

ItsRizee committed:
Product page update
Added: Input fields for tags and timeline
Added: Image, details and timeline to product page
Added: theme selection in the nav bar
Updated: product card's design and details
Updated: catalog page

ItsRizee committed:
Fixing bugs
Fixed: theme toggle
Fixed: design of website
Fixed: fitting of images in product card
Removed: unused theme attribute in database and user object

ItsRizee committed:
Fixing bugs
Added: adding tags on enter
Added: check of file type with javascript
Fixed: labels for all input fields

ItsRizee committed:
Merge remote-tracking branch 'origin/main'

ItsRizee committed:
Solving merge problems

ItsRizee committed:
Requests
Added: adding new request
Added: Changing status of requests
Added: Removing requests

ItsRizee committed:
Requests
Fixed: toggle of Drawer
Fixed: getting products by query
Added: notifications with react-toastify
Removed: requests and orders fields from firestore database

ItsRizee committed:
Home page
Added: getting newProducts from DB
Added: getting trendingProducts from DB
Added: likes to products
Added: Showing interests of user in profile page

ItsRizee committed:
Requests
Fixed: how requests are deleted

Products:
Fixed: how products are unliked

ItsRizee committed:
Home page
Added: Arrows to item scroll
Added: isTouchEnabled to userStorage
Added: getting discover products from database
Removed: search bar
Updated: Product card's title and tags

ItsRizee committed:
Fixes
Fixed: avatar image to be rounded everywhere
Added: notifications for all possible errors, successes when adding new products or requests

ItsRizee committed:
Product carf
Fixed: hide likes when user is not logged

ItsRizee committed:
Product card
Fixed: unliking of products
Fixed: resizing of item scroll on removed product
Removed: interests array from users in database

ItsRizee committed:
Fixing bugs
Fixed: responsiveness of request card's modal
Fixed: the process of saving display images for products
Fixed: Disabling adding button for NewProduct and NewRequest when it is being submitted

ItsRizee committed:
Fixing bugs
Fixed: continuously adding products

ItsRizee committed:
Followers
Added: following craftsman
Added: like button in product page
Removed: catalog from database

ItsRizee committed:
Fixing bugs
Fixed: removing of tag to be on click and not with the enter key
Fixed: made followers stateful
Fixed: positioning of remove button on request card
Fixed: rendering of profile page

atanasovstefan commented on diploma-project/components/Drawer.jsx:
```diff
@@ -11,15 +12,15 @@ const Drawer = ({page_content, sidebar_content}) => {
                 <div className="bg-base-100 text-base-content h-full w-screen sm:w-96">
                     <button className="btn btn-sm btn-circle btn-ghost mt-2.5 ml-2.5" onClick={closeDrawer}>✕</button>
-                    {sidebar_content}
+                    {toggleDrawerContent === true ? <NewProduct/> : <NewRequest craftsman={craftsman}/>}
```
IMO when rendering components, the ternary operator is harder to read when it's on one line

ItsRizee committed:
Fixing bugs
Fixed: fetching new products data server-side rather than client-side

atanasovstefan commented on diploma-project/components/ItemsScroll.jsx:
```diff
@@ -1,17 +1,31 @@
+                    <div className="overflow-x-auto no-scrollbar w-auto py-2">
+                        <div className="flex space-x-5 pt-2">
+                            {listOfItems.map((item) => (
+                                <div key={item.key}>
```
You can have unexpected behaviour if you have duplicate keys

atanasovstefan commented on diploma-project/components/ItemsScrollWithArrows.jsx:
```diff
@@ -0,0 +1,96 @@
+        return () => {
+            window.removeEventListener("resize", updateCardWidth);
+        };
+    }, [listOfItems, containerRef.current]);
```
<img width="606" alt="Screenshot 2024-02-12 at 19 25 40" src="https://github.com/ItsRizee/diploma_project/assets/42142769/f52dd480-eea0-485d-987f-0d2b5a869389">

Even the linter is complaining 🙃

atanasovstefan commented on diploma-project/components/ItemsScrollWithArrows.jsx:
```diff
@@ -0,0 +1,96 @@
+            if(listOfItems.length === 0) {
+                setTranslation(null);
+            } else if (containerRef.current) {
+                const cardWidth = containerRef.current.querySelector("#item-0").getBoundingClientRect().width;
```
In React, it's generally better to avoid using IDs and instead use refs for selecting elements.

atanasovstefan commented on diploma-project/components/ProductCard.jsx:
```diff
@@ -1,36 +1,67 @@
         if (timeDifference < 24 * 60 * 60 * 1000) {
             setIsNew(true);
         }
     }, []);
```
<img width="606" alt="Screenshot 2024-02-12 at 19 32 14" src="https://github.com/ItsRizee/diploma_project/assets/42142769/de6a85db-9095-4b2e-84de-60a7f83f0a34">

atanasovstefan commented on diploma-project/components/RequestCard.jsx:
```diff
@@ -0,0 +1,94 @@
+                });
+            }
+        }
+    }, []);
```
<img width="606" alt="Screenshot 2024-02-12 at 19 36 08" src="https://github.com/ItsRizee/diploma_project/assets/42142769/6c50802c-d66f-4cfa-ab71-4f4f5f0991a5">

atanasovstefan commented on diploma-project/components/TrendingCard.jsx:
```diff
@@ -1,16 +1,23 @@
+                <p className="sm:pr-10">{product.description}</p>
                 <div className="card-actions justify-end">
-                    <button className="btn btn-primary">More</button>
+                    <button className="btn btn-primary" onClick={onClick}>More</button>
```
This is an example of what I mentioned - the onClick function navigates to a new page, but with the current implementation SEO won't work. If the onClick is used only to navigate to another page (like it's here), you should use Link

atanasovstefan commented on diploma-project/pages/catalog/[id].jsx:
```diff
@@ -1,34 +1,52 @@
+            setProducts(catalog);
         });
-    }, [user.catalog]);
+    }, []);
```
<img width="721" alt="Screenshot 2024-02-12 at 19 40 20" src="https://github.com/ItsRizee/diploma_project/assets/42142769/3f7b961a-ed22-4591-9c88-61f966945546">

You should really consider using a linter 😆

atanasovstefan commented on diploma-project/pages/catalog/[id].jsx:
```diff
@@ -40,25 +58,43 @@ const Catalog = ({user}) => {
     return (
         <div className="flex flex-col min-h-screen overflow-x-hidden">
-            <StandardLayout title="Catalog page" page_content={products ?
+            <StandardLayout title="Catalog page" craftsman={craftsman} toggleDrawerContent={toggleDrawerContent} setToggleDrawerContent={setToggleDrawerContent} page_content={products ?
```
As this is getting more complex - it's getting harder to read with the ternary operator. Maybe consider a switch or something

atanasovstefan commented on diploma-project/pages/index.jsx:
```diff
@@ -1,44 +1,73 @@
+            });
+
+            setTrendingProducts(products);
+        }, []);
```
You know the drill 😆

ItsRizee committed:
Fixing bugs
Fixed: Positioning of arrow keys in ItemsScrollWithArrows.jsx
Fixed: checking user input for new products and request
Fixed: routing to use Links for SEO
Fixed: hide product's like button from owner
Fixed: error display of signin.jsx and signup.jsx
Fixed: forbid routing to signin.jsx and signup.jsx when authenticated
Removed: Collapse from profile page

ItsRizee committed:
Fixing bugs
Fixed: Check for input for product price

ItsRizee committed:
Fixing bugs
Fixed: Check for input for product price for float numbers

ItsRizee committed:
Fixing bugs
Fixed: code quality
